sdk/program/src/nonce/state/mod.rs

```diff
@@ -74,17 +56,35 @@ impl Versions {
                     // upgrade Uninitialized legacy nonces.
                     State::Uninitialized => None,
                     State::Initialized(ref mut data) => {
-                        data.durable_nonce = DurableNonce::from_blockhash(
-                            &data.blockhash(),
-                            true, // separate_domains
-                        );
+                        data.durable_nonce = DurableNonce::from_blockhash(&data.blockhash());
                         Some(Self::Current(state))
                     }
                 }
             }
             Self::Current(_) => None,
         }
     }
+
+    /// Updates authority pubkey on the nonce account.
+    /// Returns None if the nonce account state in Uninitialized.
+    pub fn authorize(self, authority: Pubkey) -> Option<Self> {
+        let data = match self.state() {
+            State::Uninitialized => return None,
+            State::Initialized(data) => data,
+        };
```

Comment:
hasn't the caller already resolved this?

Reply:
the function is valid either way.

Reply:
I also think it would be cleaner to avoid this redundant check in my suggested approach here: https://github.com/solana-labs/solana/pull/26055/files#r903486764